_query-dsl/specialized/script-score.md

```diff
@@ -0,0 +1,354 @@
+# Script score query
+
+A `script_score` query lets you customize the score calculation by using a script. For an expensive scoring function, you can use a `script_score` query to calculate the score only for the returned documents that have been filtered.
+
+## Example
+
+For example, the following request creates an index with one document:
+
+```json
+PUT testindex1/_doc/1
+{
+  "name": "John Doe",
+  "multiplier": 0.5
+}
+```
+{% include copy-curl.html %}
+
+You can use a `match` query to return all documents that match `John`:
+
+```json
+GET testindex1/_search
+{
+  "query": {
+    "match": {
+      "name": "John"
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In the response, document 1 has the score of `0.2876821`:
+
+```json
+{
+  "took": 7,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.2876821,
+    "hits": [
+      {
+        "_index": "testindex1",
+        "_id": "1",
+        "_score": 0.2876821,
+        "_source": {
+          "name": "John Doe",
+          "multiplier": 0.5
+        }
+      }
+    ]
+  }
+}
+```
+
+Now, let's change the document score with a script that calculates the score as the value of the `_score` field multiplied by the value of the `multiplier` field. In the following query, you can access the current relevance score of a document in the `_score` variable and the `multiplier` value as `doc['multiplier'].value`:
+
+```json
+GET testindex1/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match": { 
+            "name": "John" 
+        }
+      },
+      "script": {
+        "source": "_score doc['multiplier'].value"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+In the response, the score for document 1 is half of the original score:
+
+```json
+{
+  "took": 8,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.14384104,
+    "hits": [
+      {
+        "_index": "testindex1",
+        "_id": "1",
+        "_score": 0.14384104,
+        "_source": {
+          "name": "John Doe",
+          "multiplier": 0.5
+        }
+      }
+    ]
+  }
+}
+```
+
+Optionally, you can include a `min_score` parameter that specifies a minimum threshold score for the documents to be included in the results:
+
+```json
+GET testindex1/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match": { "name": "John" }
+      },
+      "script": {
+        "source": "_score doc['multiplier'].value"
+      },
+      "min_score": 0.15
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The preceding query returns no results because the document score of the only document in the index is lower than the `min_score` threshold.
```

Comment:
I like the example here, but we might be careful promoting `min_score` use case. There's an bug related https://github.com/opensearch-project/OpenSearch/issues/8155#issuecomment-1597832932

Reply:
Removed